### ISSUE TITLE
order of transformation

### ISSUE THREAD

**Issue by yuanqing-wang** [deepmind/optax#364]
Greetings!

I couldn't find any documentation on the order of transformations in `optax.chain`. Namely, if I just wanted a Adam optimizer with L2 weight decay, should the optimizer or the `additive_weight_decay` come first?

Thanks,
yuanqing

**Comment by n2cholas** [comment]
`optax.chain` will apply operations in the order they are given in the list. For your particular example, you can see the code in `alias.py`:

https://github.com/deepmind/optax/blob/6a035251e9dc8522b74e57d736bf0abc9b962eca/optax/_src/alias.py#L275-L278

The adam scaling is applied first, then the weight decay, then finally the learning rate. Does that make sense?

**Comment by yuanqing-wang** [comment]
Thanks!
